test/tstDetailsTreeConstruction.cpp

```diff
@@ -285,23 +285,30 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(example_tree_construction, DeviceType,
   std::cout << "ref=" << ref.str() << "\n";
 
   // hierarchy generation
+  using ArborX::Box;
+  using ArborX::Details::makeLeafNode;
   using ArborX::Details::Node;
   Kokkos::View<Node *, DeviceType> leaf_nodes("leaf_nodes", n);
   Kokkos::View<Node *, DeviceType> internal_nodes("internal_nodes", n - 1);
```

Comment:
Why did the patch where you merged them together not work?

Reply:
It was a hack.  I like that we ensure that the routine to generate the hierarchy does not assume that we have that array with the internal nodes and then the leaves.